### PR TITLE
[portsorch]: Add support of cable breakout feature

### PR DESCRIPTION
**Pull request by volodymyrsamotiy**

### CODE REVIEW

File: portsyncd/portsyncd.cpp [hunk 1]
```diff
@@ -134,34 +135,70 @@ void handlePortConfigFile(ProducerStateTable &p, string file)
...
-        iss >> name >> lanes >> alias;
+        map<string, string> entry;
+
+        /* Read port configuration entry */
+        for (auto column : header)
+        {
+            iss >> entry[column];
+        }
 
         /* If port has no alias, then use its' name as alias */
```
Comment (stcheng) [inline portsyncd/portsyncd.cpp]: no apostrophe after its

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -461,6 +462,107 @@ void PortsOrch::updateDbPortOperStatus(sai_object_id_t id, sai_port_oper_status_
     }
 }
 
+bool PortsOrch::createPort(const set<int> &lane_set, uint32_t speed)
```
Comment (stcheng) [inline orchagent/portsorch.cpp]: change the name to `addPort`? it then aligns all the other add/remove functions in SwSS.

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -530,47 +644,52 @@ void PortsOrch::doPortTask(Consumer &consumer)
...
+             * 3. Initialize all ports
+             */
+            if (m_portConfigDone && (m_lanesALiasSpeedMap.size() == m_portCount))
+            {
+                for (auto it = m_portListLaneMap.begin(); it != m_portListLaneMap.end();)
+                {
+                    if (m_lanesALiasSpeedMap.find(it->first) == m_lanesALiasSpeedMap.end())
                     {
-                        SWSS_LOG_INFO("Port has already been initialized before alias:%s", alias.c_str());
+                        if (!removePort(it->first))
```
Comment (liuqu) [inline orchagent/portsorch.cpp]: Maybe it's better to redefine the input parameter of removePort. The function removePort can use port object id directly.

**Comment by nikos-li** [comment]
How did you test this and on what platforms? If you couldn't unit test on all supported platforms, then you should introduce automated tests to take care of that and make sure the feature works prior to committing.

**Comment by volodymyrsamotiy** [comment]
@nikos-li 
I performed manual tests on Mellanox platforms (we don't have acces to other platforms).
It is not possible to introduce automated tests for port breakout, because it requires changes with physical connections (split cables).

**Comment by lguohan** [comment]
@nikos-li, they will make sure the feature works for their platform. other platform owners will need to test this on their platform. Since some platform do not support certain SAI function calls, we need to make sure this feature would not break (ensure it will not be invoked on those platforms.)

**Comment by nikos-li** [comment]
@volodymyrsamotiy At minimum, you should include output from your testing and since you don't have access to other platforms, use the google unit test framework to test the code you introduced in the orch agent.

**Comment by nikos-li** [comment]
@lguohan The code in orch agent is platform independent and not under the responsibility of any platform owners. To make sure that the feature doesn't break and the code is working properly, if automated framework tests are not acceptable, then google unit test should be introduced.

**Pull request closed without merging**